File: src/main/java/org/usfirst/frc/team4571/robot/Robot.java
package org.usfirst.frc.team4571.robot;

import edu.wpi.first.wpilibj.TimedRobot;
import edu.wpi.first.wpilibj.command.Command;
import edu.wpi.first.wpilibj.command.Scheduler;
import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;

public class Robot extends TimedRobot {
    private final SendableChooser<Command> autoChooser
            = new SendableChooser<>();
    private Command autoCommand;

    public void robotInit() {
        SmartDashboard.putData("Auto Chooser", autoChooser);
    }

    public void disabledInit() {
        Scheduler.getInstance().removeAll();
    }

    public void disabledPeriodic() {
        Scheduler.getInstance().run();
    }

    public void autonomousInit() {
        autoCommand = autoChooser.getSelected();
        if (autoCommand != null) {
            autoCommand.start();
        }
    }

    public void autonomousPeriodic() {
        Scheduler.getInstance().run();
    }

    public void teleopInit() {
        if (autoCommand != null) {
            autoCommand.cancel();
        }
    }

    public void teleopPeriodic() {
        Scheduler.getInstance().run();
    }
}

File: src/main/kotlin/org/usfirst/frc/team4571/robot/Robot.kt
package org.usfirst.frc.team4571.robot

import edu.wpi.first.wpilibj.TimedRobot
import edu.wpi.first.wpilibj.command.Command
import edu.wpi.first.wpilibj.command.Scheduler
import edu.wpi.first.wpilibj.smartdashboard.SendableChooser
import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard
import org.usfirst.frc.team4571.robot.commands.autonomous.FollowPath
import org.usfirst.frc.team4571.robot.commands.autonomous.TurnCommand
import org.usfirst.frc.team4571.robot.commands.teleop.TeleOpDrive

class Robot : TimedRobot(Constants.ROBOT_PERIOD) {
    private val autoChooser = SendableChooser<Command>()
    private var autoCommand: Command? = null

    companion object {
        val leftStick = DriveStick(Constants.Controllers.LEFT_STICK)
        val rightStick = DriveStick(Constants.Controllers.RIGHT_STICK)
    }

    override fun robotInit() {
        autoChooser.addOption("Turn 90 Degrees", TurnCommand(90.0))
        autoChooser.addOption("Run test path", FollowPath("testpath"))
        SmartDashboard.putData("Auto Chooser", autoChooser)
    }

    override fun disabledInit() = Scheduler.getInstance().removeAll()

    override fun disabledPeriodic() = Scheduler.getInstance().run()

    override fun autonomousInit() {
        autoCommand = autoChooser.selected
        autoCommand?.start()
    }

    override fun autonomousPeriodic() = Scheduler.getInstance().run()

    override fun teleopInit() {
        autoCommand?.cancel()
        Scheduler.getInstance().add(TeleOpDrive)
    }

    override fun teleopPeriodic() {
        Scheduler.getInstance().run()
    }
}